expression/integration_test.go

```diff
@@ -4135,6 +4135,26 @@ func (s *testIntegrationSuite) TestDecimalConvertToTime(c *C) {
 	tk.MustQuery("select * from t").Check(testkit.Rows("2001-01-01 10:00:00.123456 2011-07-07 10:11:12"))
 }
 
+func (s *testIntegrationSuite) TestIssue9732(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	defer s.cleanEnv(c)
+
+	tk.MustQuery(`select monthname(str_to_date(null, '%m')), monthname(str_to_date(null, '%m')),
```

Comment:
could you also add some test which disables NoZeroDateMode?

Reply:
updated.